tests/test_accountant.py

```diff
@@ -30,3 +30,25 @@ def test_analytic_solution(self):
         delta_exact = compute_delta_exact(4, 10000, 100.0)
         assert delta_upper == pytest.approx(delta_exact, rel=1e-3)
         assert delta_lower == pytest.approx(delta_exact, rel=1e-3)
+
+    def test_invariance_max_compositions(self):
```

Comment:
This is a great test on how the new logic is integrated in the accountant.

As we discussed, it would be good to have a localised test on the added logic of rounding (and asserting this is working correctly).